cmake/modules/FindZeroMQ.cmake

```diff
@@ -62,6 +62,9 @@ find_path(ZeroMQ_INCLUDE_DIR NAMES "zmq.h" "zmq_utils.h"
     DOC "ZeroMQ include directories"
     NO_DEFAULT_PATH
 )
+if(NOT ZeroMQ_INCLUDE_DIR)
```

Comment:
Hi, I am going to do the comment. You propose to:

* make a _single_ call to `find_*`
* by simply dropping the `NO_DEFAULT_PATH`

The final result would, however, be [as per the quite convoluted documentation](https://cmake.org/cmake/help/v3.4/command/find_library.html) (but I have tested it and can confirm), that **the system one will always have priority** .

The reason behind the double `find_*` calls (one with `NO_DEFAULT_PATH` and `PATHS`, the second only if nothing's found and only on system's paths) is that we want **our ZeroMQ version to always have priority, and fallback on the system one if only if a custom one is not found**.

Personally, I think that this should be CMake's behavior and shouldn't be worked around like this...

A new commit with the comment is coming :-)